lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -335,6 +344,15 @@ export class UserAgentApplication {
     acquireTokenRedirect(userRequest: AuthenticationParameters): void {
         // validate request
         const request: AuthenticationParameters = RequestUtils.validateRequest(userRequest, false, this.clientId, Constants.interactionTypeRedirect, this.redirectCallbacksSet);
+
+        const requestCorrelationId = userRequest.correlationId || CryptoUtils.createNewGuid();
+        const apiEvent: ApiEvent = new ApiEvent(requestCorrelationId, this.logger);
+        apiEvent.apiEventIdentifier = API_EVENT_IDENTIFIER.AcquireTokenRedirect;
+        apiEvent.apiCode = API_CODE.AcquireTokenRedirect;
+        this.telemetryManager.startEvent(apiEvent);
+        this.telemetryManager.stopEvent(apiEvent);
+        this.telemetryManager.flush(requestCorrelationId);
```

Comment:
Can you pull out any blocks like this into helper functions somewhere?

Reply:
+1. We worked to simplify all public APIs and it helps to declutter and manage helper functions.

Reply:
@jasonnutter  we paired on this and resolved